bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptEngineFactory.java

```diff
@@ -16,44 +16,45 @@
 import java.util.Map;
 
 import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.automation.module.script.internal.provider.ScriptModuleTypeProvider;
 
 /**
- * This class is used by the ScriptManager to load ScriptEngines.
- * This is meant as a way to allow other OSGi bundles to provide custom Script-Languages with special needs (like
- * Nashorn, Groovy, etc.)
- *
- * @author Simon Merschjohann
+ * This class is used by the ScriptEngineManager to load ScriptEngines. This is meant as a way to allow other OSGi
+ * bundles to provide custom script-languages with special needs, e.g. Nashorn, Jython, Groovy, etc.
  *
+ * @author Simon Merschjohann - Initial contribution
+ * @author Scott Rushworth - added/changed methods and parameters when implementing {@link ScriptModuleTypeProvider}
  */
+@NonNullByDefault
 public interface ScriptEngineFactory {
 
+    final static ScriptEngineManager engineManager = new ScriptEngineManager();
+
     /**
-     * @return the list of supported language endings e.g. py, jy
+     * This method returns a list of file extensions and MimeTypes that are supported by the ScriptEngine, e.g. py,
+     * application/python, js, application/javascript, etc.
+     *
+     * @return List of supported script types
      */
-    List<String> getLanguages();
+    List<String> getScriptTypes();
 
     /**
-     * "scopes" new values into the given ScriptEngine
+     * This method "scopes" new values into the given ScriptEngine.
      *
      * @param scriptEngine
      * @param scopeValues
      */
     void scopeValues(ScriptEngine scriptEngine, Map<String, Object> scopeValues);
 
     /**
-     * created a new ScriptEngine
-     *
-     * @param fileExtension
-     * @return
-     */
-    ScriptEngine createScriptEngine(String fileExtension);
-
-    /**
-     * checks if the script is supported. Does not necessarily be equal to getLanguages()
+     * This method creates a new ScriptEngine based on the supplied file extension or MimeType.
      *
-     * @param fileExtension
-     * @return
+     * @param scriptType a file extension (script) or MimeType (ScriptAction or ScriptCondition)
+     * @return ScriptEngine
      */
-    boolean isSupported(String fileExtension);
+    ScriptEngine createScriptEngine(String scriptType);
```

Comment:
I still don't understand why the return value is not annotated with `@Nullable`.

What will be returned if i call `createScriptEngine("someVeryStupidString");`

As the method contains `@NonNullByDefault` and you do not annotate the return type, you state that there is no code path that will result into a return of `null`.
If a non existing / unknown `scriptType` is used, do you throw an `IllegalArgumentException` instead of returning `null`? But then it needs to be commented, too.

Reply:
Done for now, but there's are a lot of places to fix up the null annotations in automation!